session/session.go

```diff
@@ -2236,6 +2237,49 @@ func (s *session) ExecuteStmt(ctx context.Context, stmtNode ast.StmtNode) (sqlex
 	return recordSet, nil
 }
 
+func copyAndPrintRecordSet(ctx context.Context, rs sqlexec.RecordSet, vars *variable.SessionVars,
+	sql string) (sqlexec.RecordSet, error) {
+	fieldTypes := make([]*types.FieldType, len(rs.Fields()))
+	for i, field := range rs.Fields() {
+		fieldTypes[i] = &field.Column.FieldType
+	}
+
+	chk := rs.NewChunk(nil)
+	var records []chunk.Row
+	var resultRows []string
+	for {
+		if err := rs.Next(ctx, chk); err != nil {
+			return nil, err
+		}
+		if chk.NumRows() == 0 {
+			break
+		}
+		// Copy and collect rows
+		for i := 0; i < chk.NumRows(); i++ {
+			row := chk.GetRow(i)
+			records = append(records, row)
+			resultRows = append(resultRows, row.ToString(fieldTypes))
+		}
+		chk = chunk.Renew(chk, 32)
+	}
+
+	// Print all rows
+	logutil.Logger(ctx).Info("statement returned result",
+		zap.Uint64("conn", vars.ConnectionID),
+		zap.String("SQL", sql),
+		zap.Uint64("start ts", vars.TxnCtx.StartTS),
+		zap.Uint64("for update ts", vars.TxnCtx.GetForUpdateTS()),
+		zap.String("rows", "("+strings.Join(resultRows, ");\n (")+")"),
+	)
+
+	copiedRs := &sqlexec.CopiedRecordSet{
+		Records:      records,
+		ResultFields: rs.Fields(),
+	}
+	rs.Close()
```

Comment:
The lifetime of `rs` is changed compared with the original version, would it introduce extra risks I'm not sure if there are implicit dependencies of it.

Or is it possible to print the results of the `Next` call so it's unnecessary to copy or clone here?